presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -192,11 +192,13 @@ public void updateMemoryUsage(long deltaMemoryInBytes)
 
     public QueryInfo getQueryInfoWithoutDetails()
     {
-        return getQueryInfo(null);
+        return getQueryInfo(Optional.empty());
     }
 
-    public QueryInfo getQueryInfo(StageInfo rootStage)
+    public QueryInfo getQueryInfo(Optional<StageInfo> rootStage)
     {
+        requireNonNull(rootStage, "rootStage is null");
```

Comment:
No need for this check, since this method will throw NPE directly to the caller when it is accessed later on. We only add these checks for constructors or other places that store the value, since otherwise it would be very hard to know where the null (or otherwise invalid) value came from.